### PR TITLE
Fix ts types

### PR DESCRIPTION
**Pull request by douglowder**
# Summary

Fix a large number of errors in TS type generation:

- Update `@types/react` and `@types/react-native`
- Make required changes in some of the components in `src/elements`

## Test Plan

After the change, `npm install` succeeds in building the package.

### CODE REVIEW

**Comment by douglowder** [comment]
@brentvatne this branch succeeds with `npm install` and `npm test`.

**Review by brentvatne — CHANGES_REQUESTED** [review]
need to add back `extract` for global properties

File: src/elements/Circle.tsx [hunk 1]
```diff
@@ -20,14 +19,8 @@ export default class Circle extends Shape<{
   render() {
     const { props } = this;
     const { cx, cy, r } = props;
-    return (
-      <RNSVGCircle
-        ref={this.refMethod}
-        {...extract(this, props)}
```
[the remark this answers is not in the document]
Comment (douglowder) [reply src/elements/Circle.tsx:26]: Ok I'll fix that and add some tests as well

[the remark this answers is not in the document]
Comment (douglowder) [reply src/elements/Circle.tsx:26]: @brentvatne I restored the `extract(this, props)` in the affected elements.  Testing with the SVG examples looks good.  I'll try to do some snapshot tests to verify that all SVGs are rendered the same as before.

**Comment by brentvatne** [comment]
any idea why the package-lock.json diff is so huge, given that we're just updating two `@types` packages?

**Comment by douglowder** [comment]
I had to delete it in order to run npm install. There were missing package files and conflicts if I didn't do that.

**Review by brentvatne — APPROVED** [review]

**Pull request merged**